### ISSUE TITLE
All lessons in one page

### ISSUE THREAD

**Issue by dbosk** [learnlatex/learnlatex.github.io#86]
It would be nice to be able to download the whole thing as a PDF. Or have every lesson after the other in one page (i.e. no "next lesson" links).

The use-case I have in mind is to upload that all-in-one-webpage or PDF to Perusall, that way my students in the class can interact with each other while reading and experimenting (e.g. highlight passages and attach questions/discussion threads to the hightlights). Perusall is clever enough to figure out sections (in most cases anyway), so that one can choose sections for assignments.

At the moment, Perusall breaks the buttons (for opening examples in OverLeaf), so a PDF without the buttons would be best at the moment. But I suppose putting all lessons in one page should be easier to implement.

**Comment by davidcarlisle** [comment]
I don't think a PDF version is really feasible as I'm not aware of any portable way to run the examples from PDF.

I doubt whether we would want to directly support making a single HTML file version however if you clone the markdown source repository it would probably not be too difficult to use a modified Jekyll configuration (or even just simply concatenate the markdown) and make your own version.

Note this is a personal response, others in the group may want to comment differently.

**Comment by josephwright** [comment]
At the moment, I'm minded to leave this open and see what our developer (@jonasjacek) things: it might be possible to arrange some 'one page' version.

I'm really not seeing how a PDF would work: the reason we are doing the work here is that the source material, which was a physical course, is not that easy to make interactive.

**Comment by dbosk** [comment]
> I don't think a PDF version is really feasible as I'm not aware of any portable way to run the examples from PDF.

Indeed, I had a "dead" version in mind for that (something along the lines of pandoc to convert markdown to PDF). But it would probably be more difficult.

> I doubt whether we would want to directly support making a single HTML file version however if you clone the markdown source repository it would probably not be too difficult to use a modified Jekyll configuration (or even just simply concatenate the markdown) and make your own version.

Having just glanced at the code, this seems more easy. But I'd have to dig into Jekyll and GitHub pages for that. So I decided to open an issue for now and attempt to solve it later when I have more time.

It's inspired by other pages, like the [Pro Git book](https://git-scm.com/book/en/v2) and [Git Magic](http://www-cs-students.stanford.edu/~blynn/gitmagic/).

> I'm really not seeing how a PDF would work: the reason we are doing the work here is that the source material, which was a physical course, is not that easy to make interactive.

Indeed, I want to add another type of interaction in my case :-)

**Comment by josephwright** [comment]
@dbosk  I think you've got to remember that we are trying to complement books, not replace them :)

**Comment by jonasjacek** [comment]
I am not really fond of the one-pager idea, at least not for the general set up (where to put the 'more' sections for each lesson?). I suggest to skip the idea for now, for the sake of findability of our contents in Search Engines, quick loading times and digestible, short contents.

We could add it at a later stage, but prevent crawling from Search Engines - just for users. It will be duplicate content though, which sounds like something we want to avoid.

**Comment by jonasjacek** [comment]
I stand with what I've mentioned in my previous comment. It also doesn't make sense to duplicate contents at this point, where so many contents are still not translated. Even then, I feel this is something that cannot easily be done automatically (although stuff like this exisits: https://github.com/ozum/concat-md). Heading levels will probably get mixed up and defining what is "more" and how to skip it, is not trivial.

I am strongly against that idea. Is this still an open issue?

**Comment by dbosk** [comment]
On tis  5 jan 2021 11:39:36, Jonas Jared Jacek wrote:
> I stand with what I've mentioned in my previous comment. It also 
> doesn't make sense to duplicate contents at this point, where so many 
> contents are still not translated. Even then, I feel this is something 
> that cannot easily be done automatically (although stuff like this 
> exisits: https://github.com/ozum/concat-md). Heading levels will 
> probably get mixed up and defining what is "more" and how to skip it, 
> is not trivial.
> 
> I am strongly against that idea. Is this still an open issue?

It's still an issue, but I'll probably go for a solution in a fork with 
occasional merges of updates.
